src/pudl/metadata/classes.py

```diff
@@ -711,10 +711,8 @@ def to_sql(  # noqa: C901
                 )
             elif self.type == "date":
                 checks.append(f"{name} IS DATE({name})")
-            # Need to ensure that the string representation of the datetime only
-            # includes whole seconds or this check will fail.
-            # elif self.type == "datetime":
-            #    checks.append(f"{name} IS DATETIME({name})")
+            elif self.type == "datetime":
+                checks.append(f"{name} IS DATETIME({name})")
```

Comment:
This works now that I'm providing a formatting string in the SQLite specific datetime type.